### PR TITLE
[Term Entry] JavaScript Array.prototype.findLastIndex()

### PR DESCRIPTION
**Pull request by codewithfan**
### Description

Closes Issue: #2351
Created findLastIndex folder and added content to a new markdown file findLastIndex.md.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+findIndex = age.findLastIndex((value, index) => {
+    console.log('Checking age index: ', index);
+    return value === 80
+})
+
+
+```
+
+> **Note:** If you ever need to search for an element's index within the array, and you know the value will be towards the end, using .findLastIndex shortens the iteration path.
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:80]:
```suggestion
> **Note:** If ever needing to search for an element's index within the array, and knowing the value will be towards the end, using **.findLastIndex** shortens the iteration path.
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+```shell
+9
+```
+
+## Codebyte Example
+
+```codebyte/js
+
+In this example, we will showcase the difference on how .findIndex() and .findLastIndex() iterate through the array.
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:52]:
```suggestion
```
Let's add the line before the code snippet

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:53]:
```suggestion
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+This results in the following output:
+
+```shell
+9
+```
+
+## Codebyte Example
+
+```codebyte/js
+
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:51]:
```suggestion
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+This results in the following output:
+
+```shell
+9
+```
+
+## Codebyte Example
+
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:49]:
```suggestion

In this example, showcasing the difference on how **.findIndex()** and **.findLastIndex()** iterate through the array:

```
also try not to use pronouns

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Some small changes from my side. 🙂

**Comment by SSwiniarski** [comment]
@codewithfan tag me when you're done with @dakshdeepHERE's comments and I'll do a first maintainer review.

**Comment by codewithfan** [comment]
Thank you for optimizing the code @dakshdeepHERE! 🔥  Ready for maintainer review @SSwiniarski. Thank you!

**Comment by dakshdeepHERE** [comment]
> Thank you for optimizing the code @dakshdeepHERE! 🔥 Ready for maintainer review @SSwiniarski. Thank you!

Good Luck with the maintainer's review ❤️

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
+---
+Title: '.findLastIndex()'
+Description: 'Iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, returns -1.'
```
Comment (SSwiniarski) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:3]:
```suggestion
Description: 'Iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, it returns -1.'
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Methods'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `.findLastIndex()` iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, returns -1.
```
Comment (SSwiniarski) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:15]:
```suggestion
The `.findLastIndex()` iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, it returns -1.
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+console.log(findIndexResult);
+```
+
+This results in the following output:
+
+```shell
+9
+```
+
+## Codebyte Example
```
Comment (SSwiniarski) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:48]: Unfortunately the `findLastIndex()` function isn't recognized by codebytes. So we can't have a codebyte for this entry. However, I think this is a good example, so could you revise this to be "Example 2" with the same code in a ` ```js ` block, and include an output block afterwards?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@codewithfan I did a review and had a couple of comments. Most important is we can't do a codebyte with this function, but I'd like to keep the example you used.

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md (hunk 8 above)
Comment (codewithfan) [reply content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md]: Resolved with commit ea86dbb8803f93d8e57390b0045cd3c3f18ded5c

**Comment by codewithfan** [comment]
Thank you again! @dakshdeepHERE 👍🏾  Pending your review @SSwiniarski

**Review by SSwiniarski — APPROVED** [review]
@codewithfan looks good for a second maintainer review.

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
+---
+Title: '.findLastIndex()'
+Description: 'Iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, it returns -1.'
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:3]:
```suggestion
Description: 'Iterates through the array in reverse order and returns the index that passes the provided testing function.'
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Methods'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `.findLastIndex()` iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, it returns -1.
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:15]:
```suggestion
The **`.findLastIndex()`** iterates through the array in reverse order and returns the index that passes the provided testing function. If no item passes, it returns `-1`.
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+## Syntax
+
+```pseudo
+array.findLastIndex((element, index) => {...});
+```
+
+The function can be invoked with two arguments:
+
+- `element`: The current element we are iterating through.
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:25]:
```suggestion
- `element`: The current element in the iteration.
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+- `index` (optional): The index of the array element.
+
+## Example
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:28]: I would recommend deleting this example. This is essentially a duplicate from the findLast entry and it isn't really demonstrating anything that isn't present in the second example .

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+This results in the following output:
+
+```shell
+9
+```
+
+## Example 2
+
+In this example, showcasing the difference on how **.findIndex()** and **.findLastIndex()** iterate through the array:
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:50]:
```suggestion
This example showcases the difference in the how `.findIndex()` and `.findLastIndex()` iterate through the array:
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+console.log('findIndex() method: ');
+
+let findIndex = age.find((value, index) => {
+  console.log('Checking age index: ', index);
+  return value === 80;
+});
+```
+
+```shell
+0, 1, 2, 3, 4, 5, 6, 7, 8
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:64]: This isn't quite right, it should be "Checking age index: 0" ....

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+```js
+Now we use .findLastIndex().
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:68]:
```suggestion
// Now we use .findLastIndex().
```

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+console.log('findLastIndex() method: ');
+
+findIndex = age.findLastIndex((value, index) => {
+    console.log('Checking age index: ', index);
+    return value === 80
+})
+```
+
+```shell
+6
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:81]: same edit as above

File: content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+> **Note:** If ever needing to search for an element's index within the array, and knowing the value will be towards the end, using **.findLastIndex** shortens the iteration path.
```
Comment (caupolicandiaz) [inline content/javascript/concepts/arrays/terms/findLastIndex/findLastIndex.md:84]:
```suggestion
> **Note:** If an element is known to be at the end of an array, using `.findLastIndex` can shorten the iteration path.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@codewithfan I've completed my initial review. Please update the entry, and tag me when you've finished making the changes. Thanks

**Review by codewithfan — COMMENTED** [review]
Thank you for the feedback @caupolicandiaz 👍🏾  pending your review.

**Review by caupolicandiaz — APPROVED** [review]
@codewithfan @SSwiniarski I've made a few additional small changes. Looks good to merge

**Comment by codewithfan** [comment]
Thank you for all the work you do @caupolicandiaz 👍🏾

**Comment by SSwiniarski** [comment]
@codewithfan Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/javascript/arrays/findLastIndex

image

**Pull request merged**